core/src/test/java/com/graphhopper/routing/util/NameSimilarityEdgeFilterTest.java

```diff
@@ -69,9 +69,88 @@ public void testAccept() {
 
         edge = createTestEdgeIterator("Hauptstraße");
         assertTrue(edgeFilter.accept(edge));
+
+        edge = createTestEdgeIterator("Hauptstrasse");
+        assertTrue(edgeFilter.accept(edge));
+
+        edge = createTestEdgeIterator("Hauptstr.");
+        assertTrue(edgeFilter.accept(edge));
+
+        edgeFilter = new NameSimilarityEdgeFilter(new DefaultEdgeFilter(new CarFlagEncoder()), "Rue Notre-Dame O Montréal");
+        edge = createTestEdgeIterator("Rue Dupré");
+        assertFalse(edgeFilter.accept(edge));
+
+        edge = createTestEdgeIterator("Rue Notre-Dame Ouest");
+        assertTrue(edgeFilter.accept(edge));
+
+        edgeFilter = new NameSimilarityEdgeFilter(new DefaultEdgeFilter(new CarFlagEncoder()), "227 Rue Saint-Antoine O, Montréal");
+        edge = createTestEdgeIterator("Rue Saint-Antoine O");
+        assertTrue(edgeFilter.accept(edge));
+
+        edge = createTestEdgeIterator("Rue Saint-Jacques");
+        assertFalse(edgeFilter.accept(edge));
+
+        edgeFilter = new NameSimilarityEdgeFilter(new DefaultEdgeFilter(new CarFlagEncoder()), "1025 Rue de Bleury, Montréal, QC H2Z 1M7");
+        edge = createTestEdgeIterator("Rue de Bleury");
+        assertTrue(edgeFilter.accept(edge));
+
+        edge = createTestEdgeIterator("Rue Balmoral");
+        assertFalse(edgeFilter.accept(edge));
+
+        // Modified Test from Below
+        edgeFilter = new NameSimilarityEdgeFilter(new DefaultEdgeFilter(new CarFlagEncoder()), "257 Main Road, Claremont, Cape Town, 7708, Afrique du Sud");
+        edge = createTestEdgeIterator("Main Road");
+        assertTrue(edgeFilter.accept(edge));
+
+        edgeFilter = new NameSimilarityEdgeFilter(new DefaultEdgeFilter(new CarFlagEncoder()), "Cape Point Rd, Cape Peninsula, Cape Town, 8001, Afrique du Sud");
+        edge = createTestEdgeIterator("Cape Point / Cape of Good Hope");
+        assertTrue(edgeFilter.accept(edge));
+
+        edgeFilter = new NameSimilarityEdgeFilter(new DefaultEdgeFilter(new CarFlagEncoder()), "Viale Puglie, 26, 20137 Milano, Italy");
+        edge = createTestEdgeIterator("Viale Puglie");
+        assertTrue(edgeFilter.accept(edge));
+
+
+
+
+
+
+
+
     }
 
-    @Test
+    @Ignore
```

Comment:
The problem with this PR are these examples. I used Google Maps to get Addresses and compared them to the OSM Address. The differences are very big sometimes.